### PR TITLE
:art: feat: Document block redesign block redesign #640

### LINKED ISSUE #640: :lipstick:  Document Block Redesign: block redesign
## User stories

As a user, I want the "Document Block" component in the app to be redesigned to enhance its usability and visual appeal.

---

Quick links: [Summary](#summary) | [Description](#description) | [Template](#template) | [Example](#example) | [Resources](#resources)

---

### Technical Description
The current design is: 
image

It should be changed to:
image

The files can be found at `libs/features/convs-mgr/stories/blocks/library/document-message-block/`. 

The designs are [here](https://www.figma.com/file/ZGHzSstXOYPMFU1EiqjEHD/Untitled?type=design&node-id=1%3A4&mode=design&t=Hf7DxaUUMQ3BWy57-1)

### Acceptance Criteria

1. The block should look exactly like the designs.
2. Use of `rem`, `em` as opposed to `px`.
3. All changes should be isolated to the following  `libs/features/convs-mgr/stories/blocks/library/document-message-block/`.

### PR DESCRIPTION
**Pull request by thadeusdev**
Fixes #640

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)e

# Screenshot (optional)
Screenshot from 2023-11-03 00-18-56

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] My changes generate no new warnings

### CODE REVIEW

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/document-block.jsplumb.ts [hunk 1]
```diff
@@ -20,10 +20,10 @@ export function _DocumentBlockDecoratePlumb(block: DocumentMessageBlock, comp: C
   jsPlumb.addEndpoint(comp.location.nativeElement, {
     // Whether the anchor is target (Other Block -> This Block)
     target: true,
-    cssClass:"block_endpoint",
-    endpoint: "Dot",
-    anchor: "Left",
-    maxConnections: -1
+    cssClass: 'block_endpoint',
+    endpoint: 'Dot',
+    anchor: [0, 0.11, 0, 1],
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/document-block.jsplumb.ts:25]: Lets leave this as 'Left'

File: libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.html [hunk 2]
```diff
@@ -1,31 +1,69 @@
...
+    [formGroup]="documentMessageForm"
+    fxLayout="column"
+    fxLayoutALign="start center"
+  >
+    <div
+      [id]="docInputId"
+      class="doc_bg"
+      [ngClass]="byPassedLimits.length ? 'error' : ''"
+    >
+      <div
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.html:12]: fix your indents.

File: libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.ts [hunk 3]
```diff
@@ -54,25 +66,40 @@ export class DocumentBlockComponent implements OnInit, OnDestroy {
...
-    if (this.byPassedLimits.find(limit => limit.platform === "WhatsApp")) this.whatsappLimit = true;
-    else if (this.byPassedLimits.find(limit => limit.platform === "messenger")) this.messengerLimit = true;
+    this.byPassedLimits = this._docUploadService.checkFileSizeLimits(
+      fileSize,
+      'document'
+    );
+
+    if (this.byPassedLimits.find((limit) => limit.platform === 'WhatsApp'))
+      this.whatsappLimit = true;
+    else if (
```
Comment (LemmyMwaura) [inline libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.ts:92]: same here. Don't indent files you haven't worked on. This destroys the git history as it makes it seem like you worked on this. Just something to keep in mind

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]
sync your changes and then resolve conflicts.

File: libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.html (hunk 2 above)
Comment (thadeusdev) [reply libs/features/convs-mgr/stories/blocks/library/document-message-block/src/lib/components/document-block/document-block.component.html:12]: fixed this. thanks for the heads up

File: libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/document-block.jsplumb.ts (hunk 1 above)
Comment (thadeusdev) [reply libs/features/convs-mgr/stories/blocks/library/main/src/lib/providers/document-block.jsplumb.ts]: noted. thank you the heads up

**Comment by LemmyMwaura** [comment]
Solve conflicts

**Comment by LemmyMwaura** [comment]
Closing this as it's been resolved.

**Pull request closed without merging**